addons/account/models/partner.py

```diff
@@ -293,6 +293,12 @@ class ResPartner(models.Model):
     _name = 'res.partner'
     _inherit = 'res.partner'
 
+    company_registry_number = fields.Char(string="Company ID", help="The registry number of your company. Use it if different from the Tax ID and as a unique identifier in the database.")
```

Comment:
> of your company

?

> Use it if different from the Tax ID and as a unique identifier in the database.

Not very clear for a user, it is also missing a verb

Reply:
Doesn't this field make `company_registry` useless in `base` on `res.company`? (or shouldn't it be a related or something?)

Reply:
> 

"Use" is also a verb, but I guess we could simplify by saying "This is must be unique by partner".

Reply:
Another concern is about partner hierarchy (multiple contacts/addresses/...). Shouldn't this field be only visible on the "main"/commercial partner?

Reply:
> Use it if it is different from the Tax ID and has a unique identifier in the database. 
I'm also not sure why we are talking about some constraint here? It has nothing to do with the purpose of the field (and if it is not unique by definition, why do we put a constraint?)

I modified the help to: "The registry number of the company. Use it if it is different from the Tax ID. It must be unique across all partners of a same country". I hope this makes the field clearer (and also explains the need for the constraint) ?


> Another concern is about partner hierarchy (multiple contacts/addresses/...). Shouldn't this field be only visible on the "main"/commercial partner?

Also extended `_commercial_fields` such that this field is added to the ones "managed by the commercial entity to which a partner belongs" which should fix your last remark.

As discussed IRL with @william-andre, we'll move this field from `res_company` to `res_partner` and make it related in `res_company`

Reply:
> Also extended _commercial_fields such that this field is added to the ones "managed by the commercial entity to which a partner belongs" which should fix your last remark.

It also needs to be invisible in the view of partners with a `parent_id`